doozer/doozerlib/cli/scan_sources.py

```diff
@@ -54,6 +68,143 @@ def run(self):
+        reconciled = False
+
+        # Attempt a fast-forward merge
+        rc, _, _ = exectools.cmd_gather(cmd=['git', 'pull', '--ff-only', 'public_upstream', branch_name])
+        if not rc:
+            # fast-forward succeeded, will push to openshift-priv
+            self.runtime.logger.info('Fast-forwarded %s from public_upstream/%s', metadata.name, branch_name)
+            reconciled = True
+
+        else:
+            # fast-forward failed, trying a merge commit
+            rc, _, _ = exectools.cmd_gather(cmd=['git', 'merge', f'public_upstream/{branch_name}',
+                                                 '-m', f'Reconciliation with public openshift/{repo_name}'],
+                                            log_stderr=True,
+                                            log_stdout=True)
+            if not rc:
+                # merge succeeded, will push to openshift-priv
+                reconciled = True
+                self.runtime.logger.info('Merged public_upstream/%s into %s', branch_name, metadata.name)
+
+        if not reconciled:
+            # Could not rebase from public upstream: need manual reconciliation. Log a warning and return
+            self.runtime.logger.warning('failed rebasing %s from public upstream: will need manual reconciliation',
+                                        metadata.name)
+            self.issues.append({'name': metadata.distgit_key,
+                                'issue': 'Could not rebase into -priv as it needs manual reconciliation'})
+            return
+
+        if self.dry_run:
+            self.runtime.logger.info('Would have tried reconciliation for %s/%s', repo_name, branch_name)
+            return
+
+        # Try to push to openshift-priv
+        try:
+            exectools.cmd_assert(
+                cmd=['git', 'push', 'origin', branch_name],
+                retries=3)
+
+            # Successfully pushed: mark repo as changed
+            message = f'{metadata.name} reconciled with public openshift/{repo_name}'
+            rebuild_hint = RebuildHint(RebuildHintCode.REBASE_INTO_PRIV, message)
+            self.runtime.logger.info(message)
+
+            if metadata.meta_type == 'rpm':
+                self.add_assessment_reason(metadata, rebuild_hint)
+                self.changing_rpm_metas.add(metadata)
+
+            elif metadata.meta_type == 'image':
+                self.add_image_meta_change(metadata, rebuild_hint)
+
+            else:
+                raise IOError(f'Unsupported meta type: {metadata.meta_type}')
+
+        except ChildProcessError:
+            # Failed pushing to openshift-priv
+            self.runtime.logger.warning('failed pushing to openshift-priv for %s', metadata.name)
+            self.issues.append({'name': metadata.distgit_key,
+                                'issue': 'Failed pushing to openshift-priv'})
+
+    def _do_shas_match(self, public_org, priv_org, repo_name, branch_name) -> bool:
```

Comment:
We can't assume the branch names are the same in public and private repo. Let's separate them:
```
    def _do_shas_match(self, public_org, public_repo_name, public_branch_name, priv_org, priv_repo_name, priv_branch_name) -> bool:
```